src/modules/Packages.rb

```diff
@@ -1103,9 +1103,11 @@ def ComputeSystemPatternList
         end
       end
 
-      # FATE #302116
-      # BNC #431580
+      # FATE #302116, BNC #431580
+      # select both mandatory and optional patterns
       required_patterns = PackagesProposal.GetAllResolvables(:pattern)
+      required_patterns.concat(PackagesProposal.GetAllResolvables(:pattern, optional: true))
```

Comment:
I worry this need some renaming. It is very confusing to call it required, when it include also optional.